### PR TITLE
 [Drools-4915] fix for CVE-2019-10219

### PR DESCRIPTION
**Pull request by desmax74**
Fix CVE-2019-10219

@mareknovotny 
@jiripetrlik 
@mariofusco 
@danielezonca 

related to https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1157

### CODE REVIEW

**Comment by jiripetrlik** [comment]
@mareknovotny 
Can you please help us with review of this? Shouldn't we rather align with kie-parent etc? What do you think?

**Comment by desmax74** [comment]
@mareknovotny is present in the hacep because is used in the fabric8 (k8s client) but is not declared in its pom, sadly I've discovered this requirement debugging some silent fails during the use of the hacep.

**Comment by mareknovotny** [comment]
@desmax74 
What is the reason to define here duplicated deps we have managed in KIE-parent?
We should always use these dep managed there until there is strong reason to diverge from selectively given dependencies.

File: integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public void processMessagesAsLeaderAndCreateSnapshotTest() {
                 events = events + eventsRecords.count();
                 int attemptNumber = attempts.incrementAndGet();
                 logger.warn("Attempt number on events topic:{}", attemptNumber);
-                if(attemptNumber == 11){
+                if(attemptNumber == 30){
```
Comment (jiripetrlik) [inline integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java]: Is this change related to the version of hibernate validator? If not please create a separate PR for that and describe why is it useful.

File: integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ public void processMessagesAsLeaderAndCreateSnapshotTest() {
                 events = events + records.count();
                 int attemptNumber = attempts.incrementAndGet();
                 logger.warn("Attempt number on control topic:{}", attemptNumber);
-                if(attemptNumber == 11){
+                if(attemptNumber == 30){
```
Comment (jiripetrlik) [inline integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java]: Same as above.

File: integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java [hunk 3]
```diff
@@ -103,7 +103,7 @@ public void processMessagesAsLeaderAndCreateSnapshotTest() {
                 });
                 int attemptNumber = attempts.incrementAndGet();
                 logger.warn("Attempt number on snapshot topic:{}", attemptNumber);
-                if(attemptNumber == 11){
+                if(attemptNumber == 30){
```
Comment (jiripetrlik) [inline integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java]: Same as above.

File: integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java (hunk 1 above)
Comment (desmax74) [reply integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java]: Is a workaround to see the build without a problem when jenkins is under load

File: integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java (hunk 2 above)
Comment (desmax74) [reply integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java]: Same as above

File: integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java (hunk 3 above)
Comment (desmax74) [reply integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java]: Same as above

**Comment by desmax74** [comment]
@mareknovotny the dependency here was introduced five months ago because the version provided by the parent was obsolete compared to the version used in the fabric8 plugin, and the current (6.0.13 introduced 7 months ago) is still older than the 6.0.17

**Comment by jiripetrlik** [comment]
@mareknovotny @desmax74 
And is there any reason to have old version of hibernate-validator in kie-parent? Maybe we can change version in kie-parent and then align HA-CEP with it. What do you think?

**Comment by mareknovotny** [comment]
@jiripetrlik @desmax74  right, we could upgrade kie-parent version if that is needed and the latest 6.x is compatible. that was my question why shouldn't we do that?

**Comment by desmax74** [comment]
@jiripetrlik @mareknovotny is ok for me to upgrade kie parent with 6.1.0.Final and use that version in the projects, I've changed just here because Michael asked to fix here

**Comment by desmax74** [comment]
@mareknovotny @jiripetrlik  https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1157

**Comment by desmax74** [comment]
jenkins test this

File: pom.xml [hunk 4]
```diff
@@ -159,15 +158,10 @@
       </dependency>
 
       <!-- Validation used by fabric8 client-->
-      <dependency>
-        <groupId>javax.validation</groupId>
-        <artifactId>validation-api</artifactId>
-        <version>${version.validation.api}</version>
```
Comment (mareknovotny) [inline pom.xml]: I think you should clean all refs to this too see https://github.com/kiegroup/openshift-drools-hacep/pull/70/files#diff-600376dffeb79835ede4a0b285078036R34

Comment (desmax74) [reply pom.xml]: Yep

**Pull request merged**